Testing branch week0, pull into main

SentinelAssault opened the pull request:
Michael told me to do this. Please confirm everything is ok to merge

Michaellucaschen committed:
finishing the testing Motor for the practice Robot
Sparkmax canid for the motor is 51
the button for activate the motor is kX
file changes include RobotContainer constant and the ElevatorSubsystem

Michaellucaschen committed:
Comfirm the Code for the ElevatorSubsystem works
and have the right settings on it
changes on the period and speed

Michaellucaschen committed:
Adding path planner and set point for Elevatorsub

Michaellucaschen committed:
ElevatorSubsystem update for setting points
major changes in ElevatorSubsystem file

Michaellucaschen committed:
Arm and elevator set points update(working)

Michaellucaschen committed:
Motor CanID Update

Michaellucaschen committed:
algeasubsystem motor and trigger button binding

Michaellucaschen committed:
add algae pivot, algae intake have issues

Michaellucaschen committed:
algaesubsystem finish testing setpont changes

Michaellucaschen committed:
level3 testing approve

Michaellucaschen committed:
changes for all level setpoints L1-L4

ianylu committed:
created new commands for autonomous

added test command

ianylu committed:
Rename scoreL3test to scoreL3test.java

ianylu committed:
added new named command to be called in pathplanner

Michaellucaschen committed:
added funnel subsystem and adjusted height for elevator levels

Cedric-Kurniawan committed:
L3 Setpoint Tuned

Elevator and arm setpoints were changed to allow us to score when the bumpers are flush with the reef. Only L3 has been tuned, L4 seems fine.

Cedric-Kurniawan committed:
Algae intake working, tuned L3 and L4

Cedric-Kurniawan committed:
elevtorspeed changes

Cedric-Kurniawan committed:
Adjusted setpoints for the different gear ratio

We changed from a 45:1 to a 9:1 on the lift. All setpoint values were divided by 5. DEAL values were adjusted, single L4 auton was successful. Everything was tested at Spectrum 3/12/2025.

Cedric-Kurniawan committed:
Changed the setpoints to match the gearing on the elevator

Michaellucaschen committed:
Added visuals to shuffleboard for cameras and aligning.

Michaellucaschen committed:
changes for Constants

Cedric-Kurniawan committed:
Autons Updated But Untested

There are 2 autons per Center and Left position for Red and Blue. There are also EMERGENCY autons if things go very bad.

SentinelAssault commented:
Cedric just updated autons. They are untested, so I am hesitant to merge

LesRuthven commented on src/main/java/frc/robot/Configs.java:
this looks good.   We might want to test maxAcceleration(1200) since we are using a 15:1 reduction vs the 20:1 reduction on the 5406 robot.

Cedric-Kurniawan committed:
Added Autons UNTESTED

There are now 2 autons for each of the 6 starting positions. All of which are untested. There are also 2 VERY SIMPLE Emergency autons.

LesRuthven reviewed (approved):
Thanks for getting everything into main.  I don't have a problem with the untested autonomous paaths.   They will probably get changed and updated.  Thanks, Les

Michaellucaschen committed:
path changes

Michaellucaschen committed:
LED light update

Cedric-Kurniawan committed:
Code from 3/16/2025 Victoria

Cleaned the centerlineL4Auton and made the lift faster.

Cedric-Kurniawan committed:
Commented out conflicting cde

Code added for LEDs were commented out.

LesRuthven reviewed (approved):
Thanks for updating pathplanner and getting the tuning into main.   It is good that you found our SparkFlex/SparkMax problem as well as committing our faster lift code.

Cedric-Kurniawan committed:
Final Code from Victoria 3/17

We added a side auto that works decently well, increased the lift speed, and adjusted algae intake setpoints.

Cedric-Kurniawan commented:
I committed our final code from Victoria, it includes the auton we ran in playoffs.
